File: frontend/src/interfaces/Sound.ts
import type { Creator } from "./Creator";
import type { SoundType } from "./SoundType";
import type { Rating } from "./Rating";

export interface Sound {
  ID: number;
  title: string;
  artist: string;
  sound_type_id: number;
  sound_type?: SoundType;
  creator_id: number;
  creator?: Creator;
  ratings?: Rating[];
}

export interface CreateSoundRequest {
  title: string;
  artist: string;
  sound_type_id: number;
  creator_id: number;
}

export interface UpdateSoundRequest {
  ID: number;
  title?: string;
  artist?: string;
  sound_type_id?: number;
}

File: backend/entity/sound.go
package entity

import "gorm.io/gorm"

type Sound struct {
	gorm.Model
	Title       string     `json:"title"`
	Artist      string     `json:"artist"`
	SoundTypeID uint       `json:"sound_type_id"`
	SoundType   *SoundType `gorm:"foreignKey:SoundTypeID"`
	CreatorID   uint       `json:"creator_id"`
	Creator     *Creator   `gorm:"foreignKey:CreatorID" json:"creator"`
	Ratings     []Rating   `gorm:"foreignKey:SoundID" json:"ratings"`
}
